encoding/codecv2/codecv2.go

```diff
@@ -523,3 +607,30 @@ func compressScrollBatchBytes(batchBytes []byte) ([]byte, error) {
 
 	return outbuf[:int(outbufSize)], nil
 }
+
+// decompressScrollBatchBytes decompresses the given bytes into scroll batch bytes
+func decompressScrollBatchBytes(compressedBytes []byte) ([]byte, error) {
+	// decompress data in stream and in batches of bytes, because we don't know actual length of compressed data
+	var res []byte
+	readBatchSize := 131072
+	batchOfBytes := make([]byte, readBatchSize)
+
+	r := bytes.NewReader(compressedBytes)
+	zr, err := zstd.NewReader(r)
+	if err != nil {
+		return nil, err
+	}
+	defer zr.Close()
+
+	for {
```

Comment:
maybe the for loop is not needed here? if the size of the blob remains <= 131072.

Reply:
the size of blob remains <= 131072, but size of decompressed data maybe 131072*3? if you can provide strict upperbound for the size of decompressed data (e.g. data before compressing) then I can change it and remove cycle

Reply:
got it.